pkg/gzip/gzip.go

```diff
@@ -165,3 +191,10 @@ func ExtractTarGz(owner string, fp io.Reader, dest string) (err error) {
 	}
 	return nil
 }
+
+func firstExistingParent(path string) string {
+	for _, err := os.Stat(path); err != nil; _, err = os.Stat(path) {
```

Comment:
I assumed this wouldn't happen and that we'd at least not get an error on the root destination directory, but I've changed my mind and agree with your concern. I think I should deal with this by breaking the loop if `dirname(path) == path`